### PR TITLE
Add JS timezone db for testing

### PR DESCRIPTION
**Pull request by armanbilge**
Fixes https://github.com/http4s/http4s/issues/5231. @rossabaker can you confirm?

### CODE REVIEW

**Review by rossabaker — APPROVED** [review]
Test-only libraries make me squeamish.  I'm a little worried about it masking production failures, but I don't understand Scala.js, and this is an unblocker.  I'd rather merge this and only if someone has a better idea, Do It Right.

**Comment by armanbilge** [comment]
So to provide some background (which hopefully justifies a test-only library): unlike JDK, which ships with All The Timezones out-of-the-box, JS needs these to be added explicitly (i.e. as a massive dependency). So, it's much better to let users create their own timezone DBs with exactly the timezones they want (or, just opt into everything if they so choose).

File: project/Http4sPlugin.scala [hunk 1]
```diff
@@ -357,6 +357,7 @@ object Http4sPlugin extends AutoPlugin {
     Def.setting("org.typelevel" %%% "scalacheck-effect-munit" % V.scalacheckEffect)
   lazy val scalaJavaLocalesEnUS = Def.setting("io.github.cquiroz" %%% "locales-minimal-en_us-db" % V.scalaJavaLocales)
   lazy val scalaJavaTime = Def.setting("io.github.cquiroz" %%% "scala-java-time" % V.scalaJavaTime)
+  lazy val scalaJavaTimeTzdb = Def.setting("io.github.cquiroz" %%% "scala-java-time-tzdb" % V.scalaJavaTime)
```
[the remark this answers is not in the document]
Comment (armanbilge) [reply project/Http4sPlugin.scala:360]: I think the change will have to be made upstream. But, by all means.

[the remark this answers is not in the document]
Comment (cquiroz) [reply project/Http4sPlugin.scala:360]: WDYM upstream?

**Comment by rossabaker** [comment]
If we run our tests with All The Timezones, are we at risk of masking production code that expects specific timezones to be on the classpath?  It seems to me it's safer to have an extremely conservative set of time zones on the test path, and users can add more.

But I don't know how to avoid loading the default: the code that I think is doing this is explicitly getting a UTC-zoned value.

File: project/Http4sPlugin.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (armanbilge) [reply project/Http4sPlugin.scala:360]: Ross indicated out there's a call to `ZoneId.systemDefault` in https://github.com/http4s/http4s/issues/5231#issuecomment-923189971 but I couldn't find it in http4s. I'm happy to be wrong.

**Comment by armanbilge** [comment]
That's a fair point. @cquiroz had an idea in https://github.com/http4s/http4s/issues/5231#issuecomment-923203847.

**Pull request merged**